pkg/apis/monitoring/v1/types.go

```diff
@@ -381,7 +388,8 @@ type PrometheusSpec struct {
 	// Time duration Prometheus shall retain data for. Default is '24h' if
 	// retentionSize is not set, and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
 	// (milliseconds seconds minutes hours days weeks years).
-	Retention string `json:"retention,omitempty"`
+	// +kubebuilder:default:="24h"
```

Comment:
I would add a note that time-based retention can be disabled by setting the field to "0" (but it requires a non-empty size-base retention to be effective).